src/framework/preferences/js/Builder.js

```diff
@@ -219,7 +219,7 @@ https://github.com/fluid-project/infusion/raw/main/Infusion-LICENSE.txt
             }
         },
         events: {
-            onPrefsEditorReady: null,
+            onPrefsEditorReady: "promise",
```

Comment:
I'm not sure if you've already told me this, but if so I forgot :( What does the "promise" event type do. I'm assuming it's the same as a regular event, but returns a promise. If that's the case, why not have all events return a promise?

Reply:
Sorry, this was a new facility added to the 4.x branch recently for https://issues.fluidproject.org/browse/FLUID-4883 , primarily for use with the `onCreate` event. This is a remnant left over from trying to convert the prefs framework over to this pattern more widely until I decided to back it out since I couldn't convince myself of just how many events I could be sure would fire only once, and thinking that this wouldn't offer too much value in any case since all of those code will be rewritten soon and the events will disappear. Now reverted.